backend/server/middleware/authenticator.go

```diff
@@ -136,6 +136,17 @@ func buildPolicyForUser(ctx context.Context, db *database.Connection, userID int
 	err := db.Select(&roles, sq.Select("operation_id", "role").
 		From("user_operation_permissions").
 		Where(sq.Eq{"user_id": userID}))
+
+	var userGroupIds []int64
+	err = db.Select(&userGroupIds, sq.Select("group_id").
+		From("group_user_map").
+		Where(sq.Eq{"user_id": userID}))
+
+	var groupRoles []models.UserGroupOperationPermission
+	err = db.Select(&groupRoles, sq.Select("operation_id", "role").
+		From("user_group_operation_permissions").
+		Where(sq.Eq{"group_id": userGroupIds}))
```

Comment:
Two things:
1. you're not checking the errors here
2. This should maybe be done in a transaction. A transaction is not necessarily needed here, but it'll capture errors for you, which should allow you to ignore the errors and only check if the transaction itself succeeded. It'll also prevent weird "what if someone changes permissions between db statements"-type errors.

Reply:
oh whoops good catch!

Reply:
changed